x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptKeywordMappedFieldType.java

```diff
@@ -31,27 +26,30 @@
 import org.elasticsearch.xpack.runtimefields.query.StringScriptFieldTermsQuery;
 import org.elasticsearch.xpack.runtimefields.query.StringScriptFieldWildcardQuery;
 
-import java.io.IOException;
 import java.time.ZoneId;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
 
 import static java.util.stream.Collectors.toSet;
-import static org.elasticsearch.search.SearchService.ALLOW_EXPENSIVE_QUERIES;
 
-public final class RuntimeKeywordMappedFieldType extends MappedFieldType {
+public final class ScriptKeywordMappedFieldType extends AbstractScriptMappedFieldType {
 
     private final Script script;
     private final StringScriptFieldScript.Factory scriptFactory;
 
-    RuntimeKeywordMappedFieldType(String name, Script script, StringScriptFieldScript.Factory scriptFactory, Map<String, String> meta) {
-        super(name, false, false, TextSearchInfo.NONE, meta);
+    ScriptKeywordMappedFieldType(String name, Script script, StringScriptFieldScript.Factory scriptFactory, Map<String, String> meta) {
+        super(name, script, meta);
         this.script = script;
         this.scriptFactory = scriptFactory;
     }
 
+    @Override
+    protected String runtimeType() {
+        return "keyword";
```

Comment:
use the existing constant from KeywordFieldMapper#CONTENT_TYPE?

Reply:
I'm going to use the lookup that you built instead.